pkg/mig/reconfigure/reconfigure.go

```diff
@@ -573,20 +575,16 @@ func (r *Reconfigure) setNodeLabel(key, value string) error {
 }
 
 func (r *Reconfigure) setNodeLabels(labels map[string]string) error {
-	node, err := r.clientset.CoreV1().Nodes().Get(r.ctx, r.opts.NodeName, metav1.GetOptions{})
+	patchData, err := json.Marshal(map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"labels": labels,
+		},
+	})
 	if err != nil {
-		return fmt.Errorf("failed to get node: %w", err)
-	}
-
-	if node.Labels == nil {
-		node.Labels = make(map[string]string)
-	}
-
-	for key, value := range labels {
-		node.Labels[key] = value
+		return fmt.Errorf("failed to marshal patch data: %w", err)
 	}
 
-	_, err = r.clientset.CoreV1().Nodes().Update(r.ctx, node, metav1.UpdateOptions{})
+	_, err = r.clientset.CoreV1().Nodes().Patch(r.ctx, r.opts.NodeName, types.StrategicMergePatchType, patchData, metav1.PatchOptions{})
```

Comment:
Let's use MergePatch instead of StrategicMergePatch. The motivation is to align with the `kubectl label` [behaviour](https://github.com/kubernetes/kubernetes/blob/5300466a5c8988b479a151ceb77f49dd00065c83/staging/src/k8s.io/kubectl/pkg/cmd/label/label.go#L351) 
```suggestion
	_, err = r.clientset.CoreV1().Nodes().Patch(r.ctx, r.opts.NodeName, types.MergePatchType, patchData, metav1.PatchOptions{})
```

Reply:
Done.